Documentation egg model (#115)
equinor/flownet: docs/egg_model_example.rst

==========================
The Egg model case
==========================

> The Egg model  is a synthetic reservoir model developed by `J. D. Jansen et al., 2013 <https://rmets.onlinelibrary.wiley.com/doi/pdf/10.1002/gdj3.21>`_ [1]. We used the Egg model in the Computational Geosciences group at SINTEF Digital as a test case for developing data-driven models and testing new calibration methods for history matching.
> 
> .. image:: ./_static/sintef_logo.jpg
>    :scale: 10%

Reviewer: I suggest removing the SINTEF reference here and rather create an "About FlowNet" page where Equinor, SINTEF and TNO's logos are gathered with a short story. We could also spread a few "With support of [logos]" throught the documentation (for example the home page) and maybe also on the main page of the git Repo. @anders-kiaer What do you think about that suggestion?

```suggestion
The Egg model  is a synthetic reservoir model developed by `J. D. Jansen et al., 2013 <https://rmets.onlinelibrary.wiley.com/doi/pdf/10.1002/gdj3.21>`_ [1].
```

Data files
==========

- `The Egg Model - data files  <https://data.4tu.nl/articles/The_Egg_Model_-_data_files/12707642>`_ [2]

Modifications to the simulations setup
======================================

> We have modified the Egg model input file `Egg_Model_ECL.DATA` in order to follows Flownet workflow.

Reviewer: ```suggestion
A few modifications to the original Egg model input file `Egg_Model_ECL.DATA` are required in order run Flownet.
```

> 1. COMPAT section is changed to have differences in depth between injectors and producers. With this, we avoid problems related to degenerate tetrahedron that we obtain in the triangulation procedure for the network generation.

Reviewer: ```suggestion
1. The COMPAT section is changed to have differences in depth between injectors and producers. With this, we avoid problems related to degenerate tetrahedrons that we would otherwise obtain in the triangulation procedure for the network generation.
```

::

        COMPDAT
            'INJECT1'    2*    4     7 'OPEN' 2*     0.2 	1*          0 /                    
            'INJECT2'    2*    4     7 'OPEN' 2*     0.2 	1*          0 /            
            'INJECT3'    2*    4     7 'OPEN' 2*     0.2 	1*          0 /
            'INJECT4'    2*    4     7 'OPEN' 2*     0.2 	1*          0 /
            'INJECT5'    2*    4     7 'OPEN' 2*     0.2 	1*          0 /
            'INJECT6'    2*    4     7 'OPEN' 2*     0.2 	1*          0 /
            'INJECT7'    2*    4     7 'OPEN' 2*     0.2 	1*          0 /
            'INJECT8'    2*    4     7 'OPEN' 2*     0.2 	1*          0 /
            'PROD1'      2*    1     3 'OPEN' 2*     0.2 	1*          0 / 
            'PROD2'      2*    1     3 'OPEN' 2*     0.2 	1*          0 / 
            'PROD3'      2*    1     3 'OPEN' 2*     0.2 	1*          0 / 
            'PROD4'      2*    1     3 'OPEN' 2*     0.2	1*          0 /
        /
        
        

> 2. The section PROPS is saved as an independent file `PROPS.inc`.  

Reviewer: ```suggestion
2. The section PROPS simulation keyword is saved as an independent file `PROPS.inc`.  
```

::

            DENSITY
                900 1000          1 /
            PVCDO
                400          1 1.000E-05        5        0/
            PVTW
                400          1 1.000E-05        1        0/

                

> 3. Output variables of interest have been specified in the SUMMARY section.

Reviewer: ```suggestion
3. Additional well level outputs are specified in the SUMMARY section. This information is needed to generate production "observation data" on well level that is used in the training process of FlowNet.
```

::

        SUMMARY
            FOPR
            FWPR
            FWIR
            WOPR
            'PROD1'
            'PROD2'
            'PROD3'
            'PROD4'
            /
            WWPR
            'PROD1'
            'PROD2'
            'PROD3'
            'PROD4'
            /
            WWIR
            'INJECT1' 
            'INJECT2'
            'INJECT3' 
            'INJECT4' 
            'INJECT5'
            'INJECT6'
            'INJECT7'
            'INJECT8'
            /   
            FOPT
            FWPT
            FWIT
            WLPR
            'PROD1'
            'PROD2'
            'PROD3'
            'PROD4'
            /
            WBHPH
            /
            WBHP
            /

> 4. Then you can run OPM FLOW to update input files to FlowNet

Reviewer: ```suggestion
4. Then you can run - `OPM Flow <https://opm-project.org/?page_id=19>`_ to update input files to FlowNet
```

.. code-block:: bash

>     flow /egg_model/input_file/Egg_Model_ECL.DATA

Reviewer: I'm always surprised that this model was released with a mix of lower case and upper case letters - Eclipse (on Unix) can't handle this. Did you ran into problems with Flow?

```suggestion
    flow Egg_Model_ECL.DATA
```

Author: I got no problems with Flow, but I haven't tried with Eclipse

    
    
    

Assisted history matching example
=================================

> The configuration files follow the `YAML standard <https://yaml.org/>`_.

Reviewer: ```suggestion
The following  `YAML <https://yaml.org/>`_ configuration file for FlowNet was used to train the data-driven version of the Egg model:
```

.. literalinclude:: ../examples/egg_parameters.yml
   :language: yaml
   :linenos:
   
   

> To run the assisted history matching in flownet:

Reviewer: ```suggestion
To run the assisted history matching in FlowNet:
```

.. code-block:: bash

   flowet ahm /config_file_directory/egg_parameters.yml  output_directory
   
   

Webiz visualization
===================

.. code-block:: bash

    python output_directory/generated_app/webviz_app.py

> .. image:: ./_static/webviz_results_egg_model.png

Reviewer: Here I suggest to put an inline webviz visualization, or something along those line - but you were already looking into that direction. Maybe we can't really automate that entire training for every new FlowNet release, so "static" files could be good enough. Up to you @anders-kiaer.
